Enhance `new_group` doc to mention using NCCL concurrently.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#48872 Enhance `new_group` doc to mention using NCCL concurrently.**

Using NCCL communicators concurrently is not safe and this is
documented in NCCL docs.

However, this is not documented in PyTorch and we should add documentation for
ProcessGroupNCCL so that users are aware of this limitation.

Differential Revision: [D25351778](https://our.internmc.facebook.com/intern/diff/D25351778/)

pritamdamania committed:
Enhance `new_group` doc to mention using NCCL concurrently.

Using NCCL communicators concurrently is not safe and this is
documented in NCCL docs.

However, this is not documented in PyTorch and we should add documentation for
ProcessGroupNCCL so that users are aware of this limitation.

Differential Revision: [D25351778](https://our.internmc.facebook.com/intern/diff/D25351778/)

[ghstack-poisoned]

rohan-varma reviewed (approved):
Thanks for clarifying this! I'm assuming "one process group at a time" means that collectives issued by one PG must be completely finished (i.e. not just enqueued, but actually executed by the GPU) before we can kick off a collective via another PG. Is it worth it to make this more explicit?

pritamdamania committed:
Update on "Enhance `new_group` doc to mention using NCCL concurrently."


Using NCCL communicators concurrently is not safe and this is
documented in NCCL docs.

However, this is not documented in PyTorch and we should add documentation for
ProcessGroupNCCL so that users are aware of this limitation.

Differential Revision: [D25351778](https://our.internmc.facebook.com/intern/diff/D25351778/)

[ghstack-poisoned]

pritamdamania committed:
Update on "Enhance `new_group` doc to mention using NCCL concurrently."


Using NCCL communicators concurrently is not safe and this is
documented in NCCL docs.

However, this is not documented in PyTorch and we should add documentation for
ProcessGroupNCCL so that users are aware of this limitation.

Differential Revision: [D25351778](https://our.internmc.facebook.com/intern/diff/D25351778/)

[ghstack-poisoned]
